### ISSUE TITLE
Strange indices used in multisegment well code.

### ISSUE THREAD

**Issue by blattms** [OPM/opm-simulators#2702]
I keep getting unitialized value warnings in that code:
```
In file included from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/MultisegmentWell.hpp:487,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/BlackoilWellModel.hpp:53,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/ebos/eclwriter.hh:36,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/ebos/eclproblem.hh:56,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/flow/BlackoilModelEbos.hpp:27,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp:26,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/flow/flow_ebos_gasoil.cpp:28:
/home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/MultisegmentWell_impl.hpp: In member function ‘void Opm::MultisegmentWell<TypeTag>::handleAccelerationPressureLoss(int, typename Opm::MultisegmentWell<TypeTag>::Base::WellState&) const [with TypeTag = Opm::Properties::TTag::EclFlowGasOilProblem]’:
/home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/MultisegmentWell_impl.hpp:2180:47: warning: ‘*((void*)& accelerationPressureLoss +-7976)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         duneD_[seg][seg_upwind][SPres][WFrac] -= accelerationPressureLoss.derivative(WFrac + numEq);
In file included from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/MultisegmentWell.hpp:487,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/BlackoilWellModel.hpp:53,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/ebos/eclwriter.hh:36,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/ebos/eclproblem.hh:56,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/flow/BlackoilModelEbos.hpp:27,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp:26,
                 from /home/mblatt/src/dune/opm-master/opm-simulators/flow/flow_ebos_oilwater.cpp:28:
/home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/MultisegmentWell_impl.hpp: In member function ‘void Opm::MultisegmentWell<TypeTag>::handleAccelerationPressureLoss(int, typename Opm::MultisegmentWell<TypeTag>::Base::WellState&) const [with TypeTag = Opm::Properties::TTag::EclFlowOilWaterProblem]’:
/home/mblatt/src/dune/opm-master/opm-simulators/opm/simulators/wells/MultisegmentWell_impl.hpp:2178:40: warning: ‘*((void*)& accelerationPressureLoss +48)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         duneD_[seg][seg][SPres][SPres] -= accelerationPressureLoss.derivative(SPres + numEq);
```

And indeed for a 2 phase problem without polymer this looks a bit strange:
at [MultisegmentWell_impl.hpp#L2178](/OPM/opm-simulators/blob/30a4f2b34bdce4517111a5d77be73ec618ec398e/opm/simulators/wells/MultisegmentWell_impl.hpp#L2178) we are indexing `accelerationPressureLoss.derivative(4)` of a `DenseAd::Evaluation<double, 4>` and at [MultisegmentWell_impl.hpp#L2180](/OPM/opm-simulators/blob/30a4f2b34bdce4517111a5d77be73ec618ec398e/opm/simulators/wells/MultisegmentWell_impl.hpp#L2180) we are even `indexing accelerationPressureLoss.derivative(-1000+2)`. It seems like both are out of bound memory accesses. E.g. the first one acesses `Evaluation<double,4>::data_[1+4]` and the array has size 4.

But I might have overlooked something as this code is a bit away from my main level of expertise.

**Comment by totto82** [comment]
This code need some  `if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) ` guards for 2 phase problems.

**Comment by GitPaean** [comment]
Most likely, the multisegment well models has not been tested with two phase cases, so there is some development debt there.

**Comment by blattms** [comment]
Fixed with merge of #2782
